modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RapidExperimentationTest.java

```diff
@@ -0,0 +1,315 @@
+                .setApplication(application);
+        exp = createBucketsAndRunExperiment(exp, NUMBER_OF_BUCKETS_PER_EXPERIMENT);
+
+        List<Bucket> bucketList = getBuckets(exp);
+        Assignment assignment = AssignmentFactory.createAssignment().setAssignment(bucketList.get(0).label)
+                .setExperimentLabel(exp.label).setOverwrite(true);
+
+        for (int i = 1; i <= RAPID_EXP_MAX_USERS; i++) {
+            // lets do an assignment for a user
+            assignment = putAssignment(exp, assignment, new User("user" + i));
+            // lets assert the status and response code and also the state of the experiment
+            Assert.assertEquals(assignment.status, "NEW_ASSIGNMENT");
+            Assert.assertEquals(exp.state, Constants.EXPERIMENT_STATE_RUNNING);
+        }
+
+        // since the experiment crossed the maxusers the experiment should change to PUASED but it will still allow
+        // assignments to buckets
+        assignment = putAssignment(exp, assignment, new User("user6"), "PROD");
+        Assert.assertEquals(assignment.status, "NEW_ASSIGNMENT");
+        Assert.assertEquals(getExperiment(exp).state, Constants.EXPERIMENT_STATE_PAUSED);
+
+    }
+
+    /**
+     * This test case covers a scenario where we do batch assignment for a group of experiments that is a combination of
+     * rapid experiments and normal experiments. It validates the behavior when we do bulk assignment of a user to that
+     * batch
+     */
+    @Test
+    public void testAssignmentForBatchWithFewRapidExperiments() {
+
+        Page partialpage = new Page(PARTIAL_BATCH_PAGE, true);
+        // lets create experiments for the batch
+        for (int i = 1; i <= NUMBER_OF_EXPERIMENTS_PER_BATCH; i++) {
+            Experiment exp = ExperimentFactory.createExperiment("_" + UUID.randomUUID().toString(), -1)
+                    .setApplication(application);
+
+            // lets make first two experiments in the batch as rapid experiments
+            if (i <= NUMBER_OF_RAPID_EXPERIMENTS_IN_BATCH) {
+                exp.setIsRapidExperiment(true).setUserCap(RAPID_EXP_MAX_USERS);
+            }
+
+            exp = createBucketsAndRunExperiment(exp, NUMBER_OF_BUCKETS_PER_EXPERIMENT);
+            batchExperiments.add(exp);
+
+        }
+        assignPageToExperimentsList(batchExperiments, partialpage);
+
+        // the assignment will happen until the max users cap is reached for all experiments in batch
+        for (int i = 1; i <= RAPID_EXP_MAX_USERS; i++) {
+            List<Assignment> assignments = postAssignments(application, partialpage, new User("user" + i));
+            assertAssignmentResponseStatus(assignments, "NEW_ASSIGNMENT");
+        }
+
+        // if we do one more assignment the rapid experiments should change to PAUSED/STOPPED state and the remaining
+        // should be in running state
+        List<Assignment> assignments = postAssignments(application, partialpage, new User("user1000"));
+
```

Comment:
Can you also assert here on the statuses of returned assignments for those rapid experiments?